### PR TITLE
modify-reward-policy1

### PR DESCRIPTION
**Pull request by VincentQQ**
Set the reward according to the draw progress.

### CODE REVIEW

File: board.py [hunk 1]
```diff
@@ -224,6 +224,19 @@ def player_progress(self, player_id):
         return reached_checkers_num
 
 
+    def check_result (self):
+        """
+        return the reward for player one
+        """
+        if self.check_win() != 0:
```
Comment (kenziyuliu) [inline board.py]: Don't run `check_win()` twice; do  `winner = self.check_win(), if winner != 0 return winner`

File: board.py [hunk 2]
```diff
@@ -224,6 +224,19 @@ def player_progress(self, player_id):
...
+            return self.check_win()
+        player_one_progress = self.player_progress(PLAYER_ONE)
+        player_two_progress = self.player_progress(PLAYER_TWO)
+        if player_one_progress == player_two_progress:
+            return 0
+        return player_one_progress/NUM_CHECKERS \
```
Comment (kenziyuliu) [inline board.py]: use `if else`

File: board.py [hunk 3]
```diff
@@ -224,6 +224,19 @@ def player_progress(self, player_id):
         return reached_checkers_num
 
 
+    def check_result (self):
```
Comment (kenziyuliu) [inline board.py]: I think this function should be in `MCTS.py` or `train.py`; board doesnt need to know about rewards of reinforcement learning

File: MCTS.py [hunk 4]
```diff
@@ -133,8 +133,26 @@ def selfPlay(self):
 
             if num_useless_moves >= PROGRESS_MOVE_LIMIT or self.root.state.check_win():
                 break
+        return actual_play_history, self.get_reward(self.root.state)
+
+    def get_reward (self, board):
```
[the remark this answers is not in the document]
Comment (VincentQQ) [reply MCTS.py]: which line?

[the remark this answers is not in the document]
Comment (kenziyuliu) [reply MCTS.py]: def get_reward (self, board)  ---> def get_reward(self, board):

**Pull request merged**